src/dotnet/MsbuildProject.cs

```diff
@@ -16,44 +19,49 @@ internal class MsbuildProject
     {
         const string ProjectItemElementType = "ProjectReference";
 
-        public ProjectRootElement Project { get; private set; }
+        public ProjectRootElement ProjectRoot { get; private set; }
         public string ProjectDirectory { get; private set; }
 
-        private MsbuildProject(ProjectRootElement project)
+        private ProjectCollection _collection;
+        private List<NuGetFramework> _cachedTfms = null;
+        private Project _cachedEvaluatedProject = null;
+
+        private MsbuildProject(ProjectCollection collection, ProjectRootElement project)
```

Comment:
I'm using my own collection because when I haven't and used this inside the tests which was also using the collection I was getting old versions of the projects